spynnaker/pyNN/models/neuron/synaptic_matrix_app.py

```diff
@@ -0,0 +1,719 @@
+        "__use_app_keys",
+        "__matrix_size",
+        "__delay_matrix_size",
+        "__n_subatoms",
+        "__n_subedges",
+        "__syn_mat_offset",
+        "__delay_syn_mat_offset",
+        "__index",
+        "__delay_index",
+        "__received_block",
+        "__delay_received_block"
+    ]
+
+    def __init__(
+            self, synapse_io, poptable, synapse_info, app_edge,
+            n_synapse_types, all_single_syn_sz, post_vertex_slice,
+            synaptic_matrix_region, direct_matrix_region):
+        """
+
+        :param SynapseIO synapse_io: The reader and writer of synapses
+        :param MasterPopTableAsBinarySearch poptable:
+            The master population table
+        :param SynapseInformation synapse_info:
+            The projection synapse information
+        :param ProjectionApplicationEdge app_edge:
+            The projection application edge
+        :param int n_synapse_types: The number of synapse types accepted
+        :param int all_single_syn_sz:
+            The space available for "direct" or "single" synapses
+        :param ~pacman.model.graphs.common.Slice post_vertex_slice:
+            The slice of the post-vertex the matrix is for
+        :param int synaptic_matrix_region:
+            The region where synaptic matrices are stored
+        :param int direct_matrix_region:
+            The region where "direct" or "single" synapses are stored
+        """
+        self.__synapse_io = synapse_io
+        self.__poptable = poptable
+        self.__synapse_info = synapse_info
+        self.__app_edge = app_edge
+        self.__n_synapse_types = n_synapse_types
+        self.__all_single_syn_sz = all_single_syn_sz
+        self.__post_vertex_slice = post_vertex_slice
+        self.__synaptic_matrix_region = synaptic_matrix_region
+        self.__direct_matrix_region = direct_matrix_region
+
+        # Map of machine_edge to .SynapticMatrix
+        self.__matrices = dict()
+
+        # Calculate the max row info for this edge
+        self.__max_row_info = self.__synapse_io.get_max_row_info(
+            synapse_info, self.__post_vertex_slice,
+            app_edge.n_delay_stages, self.__poptable,
+            globals_variables.get_simulator().machine_time_step, app_edge)
+
+        # These are set directly later
+        self.__all_syn_block_sz = None
+        self.__app_key_info = None
+        self.__delay_app_key_info = None
+        self.__routing_info = None
+        self.__weight_scales = None
+        self.__m_edges = None
+        self.__use_app_keys = None
+
+        self.__matrix_size = (
+            self.__app_edge.pre_vertex.n_atoms *
+            self.__max_row_info.undelayed_max_bytes)
+        self.__delay_matrix_size = (
+            self.__app_edge.pre_vertex.n_atoms *
+            self.__app_edge.n_delay_stages *
+            self.__max_row_info.delayed_max_bytes)
+        vertex = self.__app_edge.pre_vertex
+        self.__n_subatoms = int(min(
```

Comment:
__n_subatoms vs __n_sub_atoms

Reply:
OK, if you want to make the code longer elsewhere, so be it...